ember-learn/guides-source
# Updating the octane branch with the last few releases

There are two tasks here to prepare for releasing Octane. One, update the octane branch with the versions released since the last time it was updated. Second, take the contents of `release` on `master` and make them be v3.14.

Here are the steps I went through.

I copied the `guides/release` of `master` into a new folder, `v3.14.0`, which is part of a normal release. 

```bash
git checkout master
git checkout -b tmp-release-3.15
mkdir guides/v3.14.0
cp -r guides/release/** guides/v3.14.0
# make the usual edits to pages.yml
git add .
git commit -m
```

Then, I merged my release branch into `octane`, discarded all changes that would have been applied to release:

```bash
git checkout octane
git checkout -b octane-release
git merge --no-ff --no-commit tmp-release-3.15
git reset -- guides/release
git checkout -- guides/release
```

Lastly, I resolved conflicts in pages like `.local.dic` and `package.json`, committed, and pushed this branch.

## How to review

- check that no changes were made to the octane files in `release` (`git diff octane --name-only`)
- check that all version directories are present in `guides`, from 1.10 to 3.14

These are the results of `git diff octane --name-only`, condensed:

```text
.local.dic
.remarkignore
.travis.yml
guides/v1.10.0/contributing/repositories.md
guides/v1.10.0/cookbook/contributing/participating_if_you_know_ember.md
guides/v1.10.0/cookbook/contributing/suggesting_a_recipe.md
guides/v1.10.0/models/the-rest-adapter.md
guides/v1.10.0/testing/index.md
guides/v1.11.0/contributing/repositories.md
guides/v1.11.0/cookbook/contributing/participating_if_you_know_ember.md
guides/v1.11.0/cookbook/contributing/suggesting_a_recipe.md
guides/v1.12.0/contributing/repositories.md
guides/v1.12.0/cookbook/contributing/participating_if_you_know_ember.md
guides/v1.12.0/cookbook/contributing/suggesting_a_recipe.md
guides/v1.13.0/contributing/repositories.md
guides/v2.0.0/contributing/repositories.md
guides/v2.1.0/contributing/repositories.md
guides/v2.10.0/contributing/repositories.md
guides/v2.10.0/routing/defining-your-routes.md
guides/v2.11.0/contributing/repositories.md
guides/v2.11.0/routing/defining-your-routes.md
guides/v2.12.0/contributing/repositories.md
guides/v2.12.0/routing/defining-your-routes.md
guides/v2.13.0/contributing/repositories.md
guides/v2.13.0/routing/defining-your-routes.md
guides/v2.14.0/contributing/repositories.md
guides/v2.14.0/routing/defining-your-routes.md
guides/v2.15.0/contributing/repositories.md
guides/v2.15.0/routing/defining-your-routes.md
guides/v2.16.0/contributing/repositories.md
guides/v2.16.0/routing/defining-your-routes.md
guides/v2.17.0/contributing/repositories.md
guides/v2.17.0/routing/defining-your-routes.md
guides/v2.18.0/contributing/repositories.md
guides/v2.18.0/routing/defining-your-routes.md
guides/v2.2.0/contributing/repositories.md
guides/v2.3.0/contributing/repositories.md
guides/v2.3.0/routing/defining-your-routes.md
guides/v2.4.0/contributing/repositories.md
guides/v2.4.0/routing/defining-your-routes.md
guides/v2.5.0/contributing/repositories.md
guides/v2.5.0/routing/defining-your-routes.md
guides/v2.6.0/contributing/repositories.md
guides/v2.6.0/routing/defining-your-routes.md
guides/v2.7.0/contributing/repositories.md
guides/v2.7.0/routing/defining-your-routes.md
guides/v2.8.0/contributing/repositories.md
guides/v2.8.0/routing/defining-your-routes.md
guides/v2.9.0/contributing/repositories.md
guides/v2.9.0/routing/defining-your-routes.md
guides/v3.0.0/contributing/repositories.md
guides/v3.0.0/routing/defining-your-routes.md
guides/v3.1.0/contributing/repositories.md
guides/v3.1.0/routing/defining-your-routes.md
guides/v3.10.0/contributing/repositories.md
guides/v3.10.0/routing/defining-your-routes.md
guides/v3.11.0/contributing/repositories.md
guides/v3.11.0/routing/defining-your-routes.md
guides/v3.12.0/routing/defining-your-routes.md
guides/v3.13.0/**
guides/v3.14.0/**
guides/v3.2.0/contributing/repositories.md
guides/v3.2.0/routing/defining-your-routes.md
guides/v3.3.0/contributing/repositories.md
guides/v3.3.0/routing/defining-your-routes.md
guides/v3.4.0/contributing/repositories.md
guides/v3.4.0/routing/defining-your-routes.md
guides/v3.5.0/contributing/repositories.md
guides/v3.5.0/routing/defining-your-routes.md
guides/v3.6.0/contributing/repositories.md
guides/v3.6.0/routing/defining-your-routes.md
guides/v3.7.0/contributing/repositories.md
guides/v3.7.0/routing/defining-your-routes.md
guides/v3.8.0/contributing/repositories.md
guides/v3.8.0/routing/defining-your-routes.md
guides/v3.9.0/contributing/repositories.md
guides/v3.9.0/routing/defining-your-routes.md
guides/versions.yml
package-lock.json
package.json
should-it-deploy-preview-app.js
```
